state/txindex/kv/kv.go

```diff
@@ -146,8 +145,12 @@ func (txi *TxIndex) indexEvents(result *abci.TxResult, hash []byte, store dbm.Ba
 
 			// index if `index: true` is set
 			compositeTag := fmt.Sprintf("%s.%s", event.Type, string(attr.Key))
+			// ensure event does not conflict with a reserved prefix key
+			if compositeTag == types.TxHashKey || compositeTag == types.TxHeightKey {
+				continue
```

Comment:
should we log error here?

Reply:
Yeah makes sense but the tx indexer doesn't have a logger. Would you want me to return an error?

Reply:
what is going to happen if we return an error? the whole batch would be discarded or the indexer would stop?

I think it would be good to somehow notify user that these keys are reserved and events containing them will be skipped. It's just a better UX.

Reply:
This is a developers error right? So they probably want to be made aware of this before putting it into production. Either by returning an error or by logging one

Reply:
developer error. returning error is fine